RecoLocalTracker/SiStripRecHitConverter/plugins/alpaka/SiStripRecHitSoA.cc

```diff
@@ -0,0 +1,258 @@
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "Geometry/CommonTopologies/interface/SimpleSeedingLayersTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/SynchronizingEDProducer.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/memory.h"
+
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "RecoLocalTracker/Records/interface/PixelCPEFastParamsRecord.h"
+
+#include "RecoLocalTracker/SiPixelRecHits/interface/PixelCPEBase.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforDevice.h"
+#include "RecoLocalTracker/SiPixelRecHits/interface/alpaka/PixelCPEFastParamsCollection.h"
+
+// #include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/Common/interface/DetSetVectorNew.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiPixelRecHitCollection.h"
+#include "DataFormats/TrackerRecHit2D/interface/SiStripMatchedRecHit2DCollection.h"
+#include "DataFormats/Math/interface/approx_atan2.h"
+
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/CommonTopologies/interface/SimplePixelTopology.h"
+#include "Geometry/CommonTopologies/interface/GluedGeomDet.h"
+
+#include "SiStripRecHitSoAKernel.h"
+#include "alpaka/mem/view/Traits.hpp"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  template <typename TrackerTraits>
+  class SiStripRecHitSoA : public stream::SynchronizingEDProducer<> {
+    using PixelBase = typename TrackerTraits::PixelBase;
+
+    using StripHits = TrackingRecHitsSoACollection<TrackerTraits>;
+    using PixelHits = TrackingRecHitsSoACollection<PixelBase>;
+
+    using StripHitsHost = TrackingRecHitHost<TrackerTraits>;
+    using PixelHitsHost = TrackingRecHitHost<PixelBase>;
+
+    using Algo = hitkernels::SiStripRecHitSoAKernel<TrackerTraits>;
+
+  public:
+    explicit SiStripRecHitSoA(const edm::ParameterSet& iConfig);
+    ~SiStripRecHitSoA() override = default;
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  private:
+    void acquire(device::Event const& iEvent, device::EventSetup const& iSetup) override {};
```

Comment:
why do you use a `SynchronizingEDProducer` if you have an empty `acquire` method ?